src/test/java/com/amihaiemil/charles/aws/AwsHttpHeadersTestCase.java

```diff
@@ -56,23 +56,33 @@ public void fetchesOriginalRequest() {
     
     /**
      * AwsHttpHeaders can perform the original {@link AwsHttpRequest}
+     */
+    @Test
+    public void performsRequest() {
+        AwsHttpHeaders<String> awsh = new AwsHttpHeaders<>(
+            new AwsHttpRequest.FakeAwsHttpRequest(),
+            new HashMap<String, String>()
+        );        
+        assertTrue(awsh.perform().equals("performed fake request"));
+        assertTrue(awsh.request().getHttpMethod().equals(HttpMethodName.POST));
+    }
+    
+    /**
+     * AwsHttpHeaders can add headers to the original {@link AwsHttpRequest}
      * @throws IOException If something goes wrong while reading
      *  the request's content.
      */
     @Test
-    public void performsRequest() throws IOException {
-    	Map<String, String> headers = new HashMap<String, String>();
+    public void addsHeaders() throws IOException {
+        Map<String, String> headers = new HashMap<String, String>();
         headers.put("Content-type", "json");
         headers.put("HeaderName", "HeaderValue");
         AwsHttpHeaders<String> awsh = new AwsHttpHeaders<>(
             new AwsHttpRequest.FakeAwsHttpRequest(), headers
```

Comment:
@SherifWaly This unit test is good, but it misses a case: the one where the original request already has some headers. We should check that AwsHttpHeaders **adds**, but **does not replace**.

So, the original request here should have a header already.
You can do it like this:

```java
        AwsHttpRequest<String> fake = new AwsHttpRequest.FakeAwsHttpRequest();
        fake.request().addHeader("previously-here", "value");
        AwsHttpHeaders<String> awsh = new AwsHttpHeaders<>(fake, headers);
```

and modify the asserts: ``received.size()`` should be 3, not 2, plus add an assert for the previous header, to see that it's still there.